web-app/src/resources/elements/panels/main-panel.js

```diff
@@ -286,31 +286,44 @@ export class MainPanel {
         if (this.dataURI) {
             let img = new Image();
             img.onload = () => {
-                if (this.options.image.fitImageToWindow) {
-
-                    let calculatedScalingFactor = 1;
-
-                    if ( (this.element.offsetWidth / img.width) < (this.element.offsetHeight / img.height)) {
-                        calculatedScalingFactor = this.element.offsetWidth / img.width;
-                    } else {
-                        calculatedScalingFactor = this.element.offsetHeight / img.height;
-                    }
-
-                    if (calculatedScalingFactor < 0.25) {
-                        this.scalingFactor = 0.125;
-                    } else if (calculatedScalingFactor < 0.5) {
-                        this.scalingFactor = 0.25;
-                    } else if (calculatedScalingFactor < 1) {
-                        this.scalingFactor = 0.5;
-                    } else {
-                        this.scalingFactor = 1;
-                    }
+                if (_.get(this.options, 'image.fitImageToWindow', false) && this.imageCanvas) {
```

Comment:
@wlin98004 really important here.  We need to use a "safe" way to get the value in case options or other parts of the object is not defined.  Also notice I do a null check on the imageCanvas (although it should be present for us by this point)